.github/workflows/test-memory-leaks.yaml

```
@@ -11,7 +11,7 @@ jobs:
           bundler-cache: true # runs 'bundle install' and caches installed gems automatically
           bundler: latest
           cache-version: v1 # bump this to invalidate cache
-      - run: sudo apt update && sudo apt install -y valgrind && valgrind --version
+      - run: sudo apt-get update && (sudo apt-get install -y valgrind || sleep 5 && sudo apt-get install -y valgrind) && valgrind --version
```

Minor: Modern versions of apt don't need `apt-get` (they're mostly aliases)

```suggestion
      - run: sudo apt update && (sudo apt install -y valgrind || sleep 5 && sudo apt install -y valgrind) && valgrind --version
```

---

While apt and apt-get do overlap in functionality, apt is not intended for script usage and makes fewer/no interface compatibility guarantees compared to underlying tools like apt-get (https://unix.stackexchange.com/questions/590699/should-i-use-apt-or-apt-get-in-shell-scripting, https://serverfault.com/questions/958003/how-to-disable-warning-apt-does-not-have-a-stable-cli-interface, https://manpages.ubuntu.com/manpages/xenial/man8/apt.8.html).

---

Interesting, thanks for sharing! I definitely did not know this, and I guess have been making the mistake of using `apt` in scripts for a while now. Oops :sweat_smile: